File: services/recommendation-service/src/main/java/com/mealmap/recommendationservice/core/dto/UserRecommendationDto.java
package com.mealmap.recommendationservice.core.dto;

import com.mealmap.recommendationservice.document.value.RecommendationItem;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "Информация о рекомендации пользователю")
public class UserRecommendationDto {
    private String userId;

    private List<RecommendationItem> items;

    private String message;

    private LocalDateTime createdAt;
}

File: frontend/src/api/recommendation/dto/UserRecommendationDto.ts
import { RecommendationItem } from "./RecommendationItem";

export interface UserRecommendationDto {
  userId: string; // UUID as a string
  items: RecommendationItem[]; // List of recommendation items
  message: string; // Optional message
  createdAt: string; // ISO date-time string
}
